plugins/woocommerce/client/blocks/assets/js/blocks/mini-cart/iapi-frontend.ts

```diff
@@ -27,73 +27,108 @@ type MiniCartContext = {
 // Inject style tags for badge styles based on background colors of the document.
 setStyles();
 
-store( 'woocommerce/mini-cart', {
-	state: {
-		get drawerOverlayClass() {
-			const { isOpen } = getContext< MiniCartContext >();
-			const baseClasses =
-				'wc-block-components-drawer__screen-overlay wc-block-components-drawer__screen-overlay--with-slide-out';
-
-			return isOpen
-				? `${ baseClasses } wc-block-components-drawer__screen-overlay--with-slide-in`
-				: `${ baseClasses } wc-block-components-drawer__screen-overlay--is-hidden`;
-		},
-
-		get badgeIsVisible() {
-			const cartHasItems = wooStoreState.totalItemsInCart > 0;
-			const { productCountVisibility } = getContext< MiniCartContext >();
+type MiniCartState = {
+	totalItemsInCart: number;
+	drawerOverlayClass: string;
+	badgeIsVisible: boolean;
+	cartIsEmpty: boolean;
+};
 
-			return (
-				productCountVisibility === 'always' ||
-				( productCountVisibility === 'greater_than_zero' &&
-					cartHasItems )
-			);
-		},
+type MiniCart = {
+	state: MiniCartState;
 
-		get cartIsEmpty() {
-			return wooStoreState.totalItemsInCart === 0;
-		},
+	callbacks: {
+		openDrawer: () => void;
+		closeDrawer: () => void;
+		overlayCloseDrawer: ( e: MouseEvent ) => void;
+	};
+};
 
-		get cartItemCount() {
-			return wooStoreState.totalItemsInCart;
-		},
-	},
+// Destructure state in an empty call to the store, to ensure that state can be correctly typed.
+const { state: miniCartState } = store< MiniCart >(
+	'woocommerce/mini-cart',
+	{},
+	{ lock: true }
+);
```

Comment:
This should not be necessary. It is a known limitation, but I think it is best resolved by typing the derived state that depends on other parts of the state. It is [documented here](https://developer.wordpress.org/block-editor/reference-guides/interactivity-api/core-concepts/using-typescript/#typing-the-derived-state).

Pushing a change to fix it.

Reply:
Fixed here: https://github.com/woocommerce/woocommerce/pull/58265/commits/a904ee0742f8d82272b2b854bf93a9465aeff45a

Reply:
Thanks that was driving me crazy, I didn't realize we could workaround it that way.